Extension/src/LanguageServer/extension.ts

```diff
@@ -1378,3 +1381,88 @@ export async function getIncludes(maxDepth: number): Promise<any> {
     const includes = await clients.ActiveClient.getIncludes(maxDepth);
     return includes;
 }
+
+// This uses several workarounds for interacting with the hover feature.
+// A proposal for dynamic hover content would help, such as the one here (https://github.com/microsoft/vscode/issues/195394)
+async function onCopilotHover(): Promise<void> {
+    if (!vscode.window.activeTextEditor) { return; }
+    // Check if the user has access to vscode language model.
+    const vscodelm = (vscode as any).lm;
+    if (!vscodelm) { return; }
+
+    // Prep hover with wait message and get the hover position location.
+    const copilotHoverResult = await clients.ActiveClient.showCopilotHover('$(loading~spin)');
+    const hoverPosition = new vscode.Position(copilotHoverResult.hoverPos.line, copilotHoverResult.hoverPos.character);
+
+    // Make sure the editor has focus.
+    await vscode.window.showTextDocument(vscode.window.activeTextEditor.document, { preserveFocus: false, selection: new vscode.Selection(hoverPosition, hoverPosition) });
```

Comment:
If the user switches active documents, it'll move the selection to an incorrect line (i.e. the line number of the previously active document), so they'll be at some random spot in the code which may not even have C++ hover, which is why I was getting only hover from GitLens in my other comment because it moved my cursor to hover over a space that had a GitLens inlay hint on it.

Reply:
No longer using the active document.